Formula/libkeccak.rb

```diff
@@ -0,0 +1,25 @@
+class Libkeccak < Formula
+  desc "Keccak-family hashing library"
+  homepage "https://github.com/maandree/libkeccak"
+  url "https://github.com/maandree/libkeccak/archive/1.2.tar.gz"
+  sha256 "a9fa976a601b749499f24975756f98c28edddfb5e6828c681ccde7cfcb95d5f8"
+
+  depends_on "make" => :build
+
+  def install
+    system "make", "install", "OSCONFIGFILE=macos.mk", "PREFIX=#{prefix}"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <libkeccak.h>
+      int main() {
+      libkeccak_spec_t spec;
+        libkeccak_state_t state;
+        return 0;
```

Comment:
This test probably needs to be improved. It's not even linking to the library. I just wanted to open this PR and see if folks from PR #32618 were interested in moving this forwards now that upstream patched the build.

CC: @danteu @nremond @cleishm